JSDoc: Add missing params to fix new ESLint errors

This is a follow-up for #22771 which introduced a new version of eslint-plugin-jsdoc. There are two rules which are more strict now:

* jsdoc/require-param
* jsdoc/check-param-name

The rules have been reduced to a warning for now with the intend to get them fixed asap since the errors are valid. #22794 is an example for an already fixed case.

Current result of `npm run lint-js`:

```
✖ 215 problems (0 errors, 215 warnings)
  0 errors and 129 warnings potentially fixable with the `--fix` option.
```

**List of files with errors:**

* [X] packages/block-editor/src/autocompleters/block.js #22975
* [X] packages/block-editor/src/components/block-navigation/tree.js #22976
* [X] packages/block-editor/src/components/block-toolbar/utils.js #22976
* [X] packages/block-editor/src/components/color-style-selector/index.js #22981
* [X] packages/block-editor/src/components/inserter/index.native.js #22981
* [X] packages/block-editor/src/components/writing-flow/index.js #22981
* [X] packages/block-library/src/navigation/block-colors-selector.js #22982
* [X] packages/block-library/src/table/edit.js #22982
* [X] packages/block-library/src/table/state.js #22982
* [x] packages/blocks/src/api/registration.js #22943
* [x] packages/blocks/src/block-content-provider/index.js #22943
* [x] packages/components/src/menu-item/index.js #22971
* [x] packages/components/src/placeholder/index.js #22971
* [x] packages/components/src/range-control/utils.js #22971
* [x] packages/components/src/toolbar-group/index.js #22971
* [x] packages/components/src/toolbar/index.js #22971
* [x] packages/components/src/tree-grid/index.js #22971
* [x] packages/components/src/tree-grid/roving-tab-index.js #22971
* [x] packages/components/src/unit-control/unit-select-control.js #22971
* [x] packages/components/src/visually-hidden/index.js #22971
* [x] packages/core-data/src/actions.js #22966
* [x] packages/e2e-test-utils/src/create-new-post.js #22948
* [x] packages/edit-post/src/components/block-settings-menu/plugin-block-settings-menu-item.js #22944
* [x] packages/edit-post/src/components/sidebar/plugin-post-status-info/index.js #22944
* [x] packages/env/lib/config.js #22911
* [x] packages/format-library/src/link/utils.js #22913
* [x] packages/interface/src/store/reducer.js #22964
* [x] packages/rich-text/src/component/boundary-style.js
* [x] packages/rich-text/src/create.js
* [x] packages/rich-text/src/get-active-formats.js
* [x] packages/rich-text/src/get-active-object.js
* [x] packages/rich-text/src/get-last-child-index.js
* [x] packages/rich-text/src/get-line-index.js
* [x] packages/rich-text/src/get-parent-line-index.js
* [x] packages/rich-text/src/get-text-content.js
* [x] packages/rich-text/src/indent-list-items.js
* [x] packages/rich-text/src/is-collapsed.js
* [x] packages/rich-text/src/is-empty.js
* [x] packages/rich-text/src/replace.js
* [x] packages/rich-text/src/to-dom.js

---

Hint from @gziolo:

> I executed `npm run lint-js -- --fix` and it adds all those missing params but they miss types and description. Still, it seems like a good way to speed up process :)

There are only two related warnings left and we can consider this PR done:

```
/Users/gziolo/Projects/gutenberg/bin/plugin/commands/performance.js
  182:0  warning  Expected @param names to be "branches, options". Got "options, branches"  jsdoc/check-param-names

/Users/gziolo/Projects/gutenberg/packages/block-editor/src/autocompleters/block.js
  32:0  warning  @param "props" does not match an existing function parameter  jsdoc/check-param-names
```

Amazing work everyone involved 👏🏻

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -121,10 +121,6 @@ module.exports = {
 					'Avoid truthy checks on length property rendering, as zero length is rendered verbatim.',
 			},
 		],
-		// Temporarily converted to warning until all errors are resolved.
-		// See https://github.com/WordPress/gutenberg/pull/22771 for the eslint-plugin-jsdoc update.
-		'jsdoc/check-param-names': 'warn',
-		'jsdoc/require-param': 'warn',
 	},
 	overrides: [
 		{

--- a/bin/plugin/commands/performance.js
+++ b/bin/plugin/commands/performance.js
@@ -205,8 +205,8 @@ async function runTestSuite( testSuite, performanceTestDirectory ) {
 /**
  * Runs the performances tests on an array of branches and output the result.
  *
- * @param {WPPerformanceCommandOptions} options Command options.
  * @param {string[]}                    branches Branches to compare
+ * @param {WPPerformanceCommandOptions} options Command options.
  */
 async function runPerformanceTests( branches, options ) {
 	// The default value doesn't work because commander provides an array.

--- a/packages/block-editor/src/autocompleters/block.js
+++ b/packages/block-editor/src/autocompleters/block.js
@@ -22,17 +22,10 @@ import BlockIcon from '../components/block-icon';
 
 const SHOWN_BLOCK_TYPES = 9;
 
-/** @typedef {import('@wordpress/block-editor').WPEditorInserterItem} WPEditorInserterItem */
-
 /** @typedef {import('@wordpress/components').WPCompleter} WPCompleter */
 
 /**
  * Creates a blocks repeater for replacing the current block with a selected block type.
- *
- * @param {Object} props                                   Component props.
- * @param {string} [props.getBlockInsertionParentClientId] Client ID of the parent.
- * @param {string} [props.getInserterItems]                Inserter items for parent.
- * @param {string} [props.getSelectedBlockName]            Name of selected block or null.
  *
  * @return {WPCompleter} A blocks completer.
  */

